R/dialogs.R

```diff
@@ -0,0 +1,110 @@
+#' Show Dialog Box
+#'
+#' Shows a dialog box with a given title and contents.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area. Contents can contain the following HTML tags:
+#'   "p", "em", "strong", "b" and "i".
+#'
+#' @param url And optional url to display under the \code{message}.
+#'
+#' @details
+#'
+#' \preformatted{
+#'     showDialog("A dialog", "Showing <b>bold</b> text in the message.")
+#' }
+#'
+#' @export
+showDialog <- function(title, message, url = NULL) {
+  callFun("showDialog", title, message, url)
+}
+
+#' Updates a Dialog Box
+#'
+#' Updates specific properties from the current dialog box.
+#'
+#' @param ... Named parameters and values to update a dialog box.
+#'
+#' @details
+#'
+#' Currently, the only dialog with support for this action is the
+#' New Connection dialog in which the code preview can be
+#' updated through this API.
+#'
+#' \preformatted{
+#'     updateDialog(code = "con <- NULL")
+#' }
+#'
+#' @export
+updateDialog <- function(...) {
+  callFun("updateDialog", ...)
+}
+
+#' Show Prompt Dialog Box
+#'
+#' Shows a dialog box with a prompt field.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area.
+#'
+#' @param default An optional character vector that fills the prompt field
+#'   with a default value.
+#'
+#' @export
+showPrompt <- function(title, message, default = NULL) {
+  callFun("showPrompt", title, message, default)
+}
+
+#' Show Question Dialog Box
+#'
+#' Shows a dialog box asking a question.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area.
+#'
+#' @param ok And optional character vector that overrides the caption for
+#'   the OK button.
+#'
+#' @param cancel An optional character vector that overrides the caption for
+#'   the Cancel button.
+#'
+#' @details
+#'
+#' @export
+showQuestion <- function(title, message, ok = NULL, cancel = NULL) {
+  callFun("showQuestion", title, message, ok, cancel)
+}
+
+#' Read Preference
```

Comment:
@jjallaire they are already namespaced with `rstudioapi_`: https://github.com/rstudio/rstudio/pull/967/files#diff-ede85aaa4cad3424f4cae655c5408f80R384 - Let me know if we prefer a different namespace and I'll make the change.